cmake/compiler/gcc/compiler_flags.cmake

```diff
@@ -120,6 +120,10 @@ if (NOT CONFIG_NEWLIB_LIBC AND
   set_compiler_property(APPEND PROPERTY nostdinc_include ${NOSTDINC})
 endif()
 
+if (NOT CONFIG_LIB_CPLUSPLUS)
+  set_compiler_property(TARGET compiler-cpp PROPERTY nostdincxx "-nostdinc++")
+endif()
```

Comment:
The compiler flags should not be checking for Kconfig settings.
Compiler flags file should simply specify what flags are defining a given compiler property.

It is the user of the compiler property to decide whether or not to apply the flag.
That is here:
https://github.com/zephyrproject-rtos/zephyr/blob/9249127456027419b8bab0418d3a5dd927e750e3/CMakeLists.txt#L1680

```suggestion
set_compiler_property(TARGET compiler-cpp PROPERTY nostdincxx "-nostdinc++")
```

Reply:
yep, that one is a bit nasty.